### ISSUE TITLE
Soft keyboard input field missing when using UnityTransparentStatusBarTheme

### ISSUE THREAD

**Issue by zoeesilcock** [Over17/UnityShowAndroidStatusBar#3]
When using the transparent theme (UnityTransparentStatusBarTheme) on my Android 7.1.1 device I lose the input field above the soft keyboard. This field is needed for Unity InputFields to work correctly. By switching to the non-transparent theme (UnityStatusBarTheme) the input field shows as expected again.

From what I can tell the difference between these two themes is the following settings:
```
<item name="android:windowTranslucentStatus">true</item>
<item name="android:windowTranslucentNavigation">true</item>
```

These two settings also change the size of the area that Unity gets to work with. With them set to true Unity gets the whole screen including the areas covered by the status and navigation bars. With them set to false Unity only gets the space between the status and navigation bars. I don't know why this would affect the input field connected to the soft keyboard.

**Comment by Over17** [comment]
Confirmed on N5 @ Marshallow and N6 @ Nougat (not a 7.1.1 specific issue).
Works in landscape mode.
Investigating...

**Comment by Over17** [comment]
Looks like an Android bug: https://code.google.com/p/android/issues/detail?id=63777
Looking into workarounds

**Comment by Over17** [comment]
So, my understanding of what's happening:
When we set system UI to be transparent, Unity view occupied the whole screen, and the transparent UI is rendered over the view.
In addition, the on-screen keyboard also becomes "transparent", and the whole input dialog is rendered in the bottom of the screen. The keyboard is however not transparent, and overlays the input dialog.

The input is still "working", and the correct string should be reported as TouchScreenKeyboard.text.


The solution seems to be to call

`getWindow().clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
getWindow().clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
`

on the SoftInputDialog's window in the Unity code.
I'll see if these changes can make it into the codebase.

Can't think of a nice workaround in the meanwhile.

**Comment by atude** [comment]
Is there any temporary method of implementing this until its properly fixed?

**Comment by Over17** [comment]
@CyberScopes362 sorry can't think of an easy one. Other than creating an own TouchScreenKeyboard or subclassing from Unity's implementation of SoftInputDialog. Or don't use the transparent theme :(

**Comment by atude** [comment]
Alright thats fine, thanks for continuing to help us with this issue!

**Comment by Over17** [comment]
I've pushed the fix to the Unity codebase. Will need some time to appear in the mainline (2017.1), then will be backported to earlier Unity versions. Check the release notes - I'll also write here.

**Comment by Over17** [comment]
Update: the fix on the Unity side released in version 2017.1b2.
Backports to older versions are on their way.

**Comment by Over17** [comment]
The fix should be available in
5.4.5p2
5.5.3p3
5.6.0p3
(and later versions).

Sorry for the long turnaround - it's taking some time to get a low-priority fix into Unity codebase, and then get it backported into older releases.

Feel free to reopen the issue if it still doesn't work for you.
